lightning/src/ln/channelmanager.rs

```diff
@@ -6657,19 +6666,8 @@ where
 						|msg, reason, err_data, phantom_ss, next_hop_unknown| {
 							log_info!(logger, "Failed to accept/forward incoming HTLC: {}", msg);
 
-							let htlc_source = HTLCSource::PreviousHopData(HTLCPreviousHopData {
-								short_channel_id: prev_short_channel_id,
-								user_channel_id: Some(prev_user_channel_id),
-								channel_id: prev_channel_id,
-								outpoint: prev_funding_outpoint,
-								counterparty_node_id: Some(prev_counterparty_node_id),
-								htlc_id: prev_htlc_id,
-								incoming_packet_shared_secret: incoming_shared_secret,
-								phantom_shared_secret: phantom_ss,
-								blinded_failure: routing.blinded_failure(),
-								cltv_expiry,
-							});
-
+							let mut prev_hop = payment.htlc_previous_hop_data();
+							prev_hop.phantom_shared_secret = phantom_ss;
```

Comment:
Perhaps add a comment why the phantom secret as set in `htlc_previous_hop_data()` isn't good.